### DISCUSSION TITLE
schema_json arguments are not documented

### DISCUSSION THREAD

**Discussion by ssbarnea** [pydantic/pydantic#2397] [Feature Request]
At this moment finding optional to give to `schema_json()` requires a lot of digging inside the implementation as official documentation does not mention any of them at its main location. Only by chance we can discover that  used of `ident=2` inside few examples.

I managed to discover that I can also give `sort_keys=True`, which is highly desired unless someone enjoys an inhuman order. If possible, I would appreciate sorting to be enabled by default.

Another aspect related to dumping schemas as that they missed the $schema field, which was weird, considering that pydantic should know what [version of the spec](https://github.com/samuelcolvin/pydantic/issues/2396) it would be following when producing them.

**Comment by PrettyWood** [comment]
Hi @ssbarnea 
Since `schema_json()` calls `json_dumps` that can be customized, we cannot directly set all the expected kwargs. You can use default `json` but also `orjson`, `ujson`, ...
I guess the documentation could be more explicit for people to look directly at the library underneath.

I feel like setting `sort_keys = True` by default would be a huge mistake since currently fields are ordered: `title`, `type`, `properties`, `required`... Setting this option we would put `properties` first, then `required`, ...

As for the version of the spec currently it's still a bit messy agreed. Most of the job at the beginning was done for `fastapi`, which tries to support OpenAPI 3.0. If they move to 3.1 (there is an open PR) I think we'll be able to be clearer and probably target https://json-schema.org/draft/2020-12/schema
